crates/bindings-csharp/Runtime/Attrs.cs

```diff
@@ -40,8 +22,65 @@ public enum ColumnAttrs : byte
     PrimaryKeyIdentity = PrimaryKeyAuto,
 }
 
-[AttributeUsage(AttributeTargets.Field, Inherited = false, AllowMultiple = false)]
-public sealed class ColumnAttribute(ColumnAttrs type) : Attribute
+/// <summary>
+/// Registers a type as the row structure of a SpacetimeDB table, enabling codegen for it.
+///
+/// <para>
+/// Multiple [Table] attributes per type are supported. This is useful to reuse row types.
+/// Each attribute instance must have a unique name and will create a SpacetimeDB table.
+/// </para>
+/// </summary>
+[AttributeUsage(AttributeTargets.Struct, AllowMultiple = true)]
+public sealed class TableAttribute : Attribute
 {
-    public ColumnAttrs Type => type;
+    /// <summary>
+    /// This identifier is used to name the SpacetimeDB table on the host as well as the
+    /// table handle structures generated to access the table from within a reducer call.
+    ///
+    /// <para>Defaults to the <c>nameof</c> of the target type.</para>
+    /// </summary>
+    public string? Name;
+
+    /// <summary>
+    /// Set to <c>true</c> to make the table visible to everyone.
+    ///
+    /// <para>Defaults to the table only being visible to its owner.</para>
+    /// </summary>
+    public bool Public = false;
+
+    public string? Index;
+
+    public IndexType? IndexType;
+
+    public string[]? IndexColumns;
 }
+
+[AttributeUsage(AttributeTargets.Field)]
+public abstract class ColumnAttribute : Attribute
+{
+    public string? Table;
+}
+
+public sealed class AutoIncAttribute : ColumnAttribute { }
+
+public sealed class PrimaryKeyAttribute : ColumnAttribute { }
+
+public sealed class UniqueAttribute : ColumnAttribute { }
+
+public sealed class IndexedAttribute : ColumnAttribute { }
+
+public static class ReducerKind
+{
+    public const string Init = "__init__";
+    public const string Update = "__update__";
+    public const string Connect = "__identity_connected__";
+    public const string Disconnect = "__identity_disconnected__";
+}
+
+[AttributeUsage(AttributeTargets.Method, Inherited = false)]
+public sealed class ReducerAttribute : Attribute
+{
+    public string? Name;
```

Comment:
Reducer kind being a special name was more of an implementation detail, and that will be changing in the new proposal (WIP in https://github.com/clockworklabs/SpacetimeDB/pull/1670) where ReducerKind is stored separately.

AFAIK the SDK proposal also doesn't allow custom `#[spacetimedb::reducer(name = "...")]`, so we shouldn't be doing that in C# either - can you please revert this to a regular attribute parameter, so it would be easier to migrate to non-name-based ReducerKind in the future.